### ISSUE TITLE
Snoopy-Logger not logging user commands from login shell

### ISSUE THREAD

**Issue by PSKyne** [a2o/snoopy#89]
Snoopy-Logger 2.4.3 installed on RHEL 6.6, `kernel: 2.6.32-573.1.1.el6.x86_64 #1 SMP`

I just updated snoopy-logger on this system because the previous version quit logging the `%{cmdline}` correctly. The log showed the following error for the `%{cmdline}`:
`lsERROR (Filter not found - i)`

After updating snoopy-logger to 2.4.3 (latest stable), `%{cmdline}` works for root and system accounts, but not for user login-accounts. The `%{cmdline}` data source only returns `bash -`. 

This breaks part of my main reason for using snoopy-logger. That is to log ALL commands executed by users & root on the system.

My snoopy.ini file is the default one installed by `snoopy-installer.sh`. Excepting I've added the following message_format line:
`message_format = "[login:%{login} uid:%{uid}-%{username} euid:%{euid}-%{eusername} sid:%{sid} tty:%{tty} cwd:%{cwd} filename:%{filename}]: %{cmdline}"`

filter_chain is set to the default (commented out, and set explicitly makes no difference)
`;filter_chain = ""`

snoopy log output follows:
`Aug  7 11:38:41 testsystemRHD2 snoopy[2703]: [login:(unknown) uid:0-root euid:0-root sid:2698 tty:(none) cwd:/ filename:/sbin/unix_chkpwd]: /sbin/unix_chkpwd testuser chkexpiry Aug  7 11:38:41 testsystemRHD2 sshd[2698]: Accepted publickey for testuser from 192.168.228.182 port 55318 ssh2 Aug  7 11:38:41 testsystemRHD2 sshd[2698]: pam_unix(sshd:session): session opened for user testuser by (uid=0) Aug  7 11:38:41 testsystemRHD2 snoopy[2705]: [login:testuser uid:501-testuser euid:501-testuser sid:2705 tty:/dev/pts/1 cwd:/home/testuser filename:/bin/bash]: -bash`

The above is the LAST entry in the log for the user until the user logs out, regardless of commands issued by the user. (eg: `ls`, `cat /etc/resolv.conf`, `pwd`, etc.)

When the user logs out, snoopy shows the sshd session of the user has been terminated.

Can someone please identify what I'm doing incorrectly?

Thank you,

PSK

**Comment by PSKyne** [comment]
Update:  It seems that snoopy-logger is logging for system accounts with UIDs lower than 500 on this system. I am seeing logging from UID:48-apache. Still, the several normal user accounts I've tried with result with nothing past their logon shell being logged.

**Comment by bostjan** [comment]
Hi PSK,

I've just updated FAQ with what I think is your issuee too. Here:
https://github.com/a2o/snoopy/blob/master/doc/FAQ.md#5-i-see-no-snoopy-output-after-initial-user-login

To check, log in as normal user and check it's environment.

Please note that Snoopy is not an auditing solution, as it is easily circumvented. It is more of a self-reassurance tool.

About cmdline input provider: what was your previous version of Snoopy, that was giving you the hard time?

b.

**Comment by PSKyne** [comment]
Thank you sir. My previous version was 2.3.6. 

I did check the user environment scripts and none of the users have any LD_PRELOADs set.

Any ideas?

**Comment by bostjan** [comment]
2.3.6 was never released. 2.2.6 perhaps?

And no, I am out of ideas, except for these:
- sth is preloading something over snoopy, or
- You are using selinux? Or some similar kernel-level hardening?

B.

**Comment by bostjan** [comment]
Also you can provide strace of snoopy-test-output command. Steps:
1. Build snoopy with --enable-dev-tools
2. Switch to regular user
3. Run command : strace /path/to/snoopy/build/dir/util/snoopy-test-output
4. Paste the strace output here

B.

**Comment by PSKyne** [comment]
Sorry BostJan, you are correct, it was 2.3.2. (sorry for the typo.)

I am using SELinux.

I will run with the dev-tools listed above and paste the output this afternoon.

Thank you,
--Patrick

**Comment by PSKyne** [comment]
Do you think this is an SELinux context issue?

**Comment by bostjan** [comment]
There is no better way of knowing than trying it out. I have no CentOS that has selinux enabled (I only use it in lxc containers running on host that does not support selinux). Try it out and see for yourself.

My guess is: something is preventing access to /dev/log socket, probably SELinux. Unfortunately I have zero experience with selinux context configuration, aside from disabling it altogether.

**Comment by PSKyne** [comment]
From what I can see by running ls -Z on /dev/log and on /usr/local/lib/libsnoopy.so*, they are all set the same. (But I'm no expert with SELinux)

-r--r--r--. root root system_u:object_r:lib_t:s0 libsnoopy.so
srw-rw-rw-. root root system_u:object_r:devlog_t:s0 /dev/log

**Comment by bostjan** [comment]
Do try with selinux disabled, to exclude this variable from analysis.

**Comment by PSKyne** [comment]
Update:

I tried on a test box that had SELinux enabled, did a clean install of 2.4.3 and user logging is working fine. Upgraded it to the latest kernel (2.6.32-573.1.1.el6) _and_ latest SELinux policies (3.7.19-279.el6) just like my production server. Both boxes have the same version of BASH and the same SELinux contexts on /etc/snoopy.ini, /usr/local/lib/libsnoopy*, /usr/local/sbin/snoopy*, /dev/log. User logging is still working on this box after upgrades of kernel & SELinux. Still doesn't work on production server. 

I've disabled SELinux on the production server (/etc/sysconfig/selinux -- SELINUX=disabled). Rebooted (I know it's not windows) and still no user logging.

On my production server, I uninstalled snoopy-logger (snoopy-disable; make uninstall; make clean), manually deleted any remaining *snoopy* files from the system and rebooted. Followed by a clean, fresh install (same as the test server), and rebooted. Still no user logging.

I'm running out of ideas. Does not appear to be related to SE Linux.
--Patrick

**Comment by bostjan** [comment]
I am out of ideas already;)

Still waiting on that strace output.

B.

**Comment by bostjan** [comment]
Was this fixed, Patrick?

**Comment by clover1983** [comment]
I have the same problem

this is the regular user with no-snoopy-output-after-initial-user-login

_____________________________________________________________

[lily@test util]$  strace /home/lily/20150922/snoopy-2.4.4/util/snoopy-test-output
execve("/home/lily/20150922/snoopy-2.4.4/util/snoopy-test-output", ["/home/lily/20150922/snoopy-2.4.4"...], [/* 23 vars */]) = 0
brk(0)                                  = 0xa85000
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6b515a4000
access("/etc/ld.so.preload", R_OK)      = 0
open("/etc/ld.so.preload", O_RDONLY)    = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=28, ...}) = 0
mmap(NULL, 28, PROT_READ|PROT_WRITE, MAP_PRIVATE, 3, 0) = 0x7f6b515a3000
close(3)                                = 0
open("/usr/local/lib/libsnoopy.so", O_RDONLY) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0PQ\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=233989, ...}) = 0
mmap(NULL, 2149952, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f6b51179000
mprotect(0x7f6b51186000, 2093056, PROT_NONE) = 0
mmap(0x7f6b51385000, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xc000) = 0x7f6b51385000
close(3)                                = 0
munmap(0x7f6b515a3000, 28)              = 0
open("/etc/ld.so.cache", O_RDONLY)      = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=36747, ...}) = 0
mmap(NULL, 36747, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f6b5159b000
close(3)                                = 0
open("/lib64/libpthread.so.0", O_RDONLY) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\340]\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=142640, ...}) = 0
mmap(NULL, 2212848, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f6b50f5c000
mprotect(0x7f6b50f73000, 2097152, PROT_NONE) = 0
mmap(0x7f6b51173000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x17000) = 0x7f6b51173000
mmap(0x7f6b51175000, 13296, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f6b51175000
close(3)                                = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6b5159a000
open("/lib64/libdl.so.2", O_RDONLY)     = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\340\r\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=19536, ...}) = 0
mmap(NULL, 2109696, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f6b50d58000
mprotect(0x7f6b50d5a000, 2097152, PROT_NONE) = 0
mmap(0x7f6b50f5a000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x2000) = 0x7f6b50f5a000
close(3)                                = 0
open("/lib64/libc.so.6", O_RDONLY)      = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0p\356\1\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=1921176, ...}) = 0
mmap(NULL, 3750152, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f6b509c4000
mprotect(0x7f6b50b4e000, 2097152, PROT_NONE) = 0
mmap(0x7f6b50d4e000, 20480, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x18a000) = 0x7f6b50d4e000
mmap(0x7f6b50d53000, 18696, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f6b50d53000
close(3)                                = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6b51599000
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6b51598000
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6b51597000
arch_prctl(ARCH_SET_FS, 0x7f6b51598700) = 0
mprotect(0x7f6b50d4e000, 16384, PROT_READ) = 0
mprotect(0x7f6b50f5a000, 4096, PROT_READ) = 0
mprotect(0x7f6b51173000, 4096, PROT_READ) = 0
mprotect(0x7f6b515a5000, 4096, PROT_READ) = 0
munmap(0x7f6b5159b000, 36747)           = 0
set_tid_address(0x7f6b515989d0)         = 15009
set_robust_list(0x7f6b515989e0, 24)     = 0
futex(0x7fff0bfdb8dc, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x7fff0bfdb8dc, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 1, NULL, 7f6b51598700) = -1 EAGAIN (Resource temporarily unavailable)
rt_sigaction(SIGRTMIN, {0x7f6b50f61c60, [], SA_RESTORER|SA_SIGINFO, 0x7f6b50f6b710}, NULL, 8) = 0
rt_sigaction(SIGRT_1, {0x7f6b50f61cf0, [], SA_RESTORER|SA_RESTART|SA_SIGINFO, 0x7f6b50f6b710}, NULL, 8) = 0
rt_sigprocmask(SIG_UNBLOCK, [RTMIN RT_1], NULL, 8) = 0
getrlimit(RLIMIT_STACK, {rlim_cur=10240*1024, rlim_max=RLIM64_INFINITY}) = 0
brk(0)                                  = 0xa85000
brk(0xaa6000)                           = 0xaa6000
open("/usr/local/etc/snoopy.ini", O_RDONLY) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=8175, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6b515a3000
read(3, ";;; REQUIRED Section\n;\n[snoopy]\n"..., 4096) = 4096
read(3, "ilable=yes) Exclude log entries "..., 4096) = 4079
read(3, "", 4096)                       = 0
close(3)                                = 0
munmap(0x7f6b515a3000, 4096)            = 0
fstat(1, {st_mode=S_IFCHR|0620, st_rdev=makedev(136, 1), ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6b515a3000
write(1, "Configuration file is enabled: /"..., 57Configuration file is enabled: /usr/local/etc/snoopy.ini
) = 57
write(1, "Configuration file found.\n", 26Configuration file found.
) = 26
write(1, "Configuration file was parsed su"..., 43Configuration file was parsed sucessfully.
) = 43
getuid()                                = 1003
getsid(0)                               = 8601
ioctl(0, SNDCTL_TMR_TIMEBASE or SNDRV_TIMER_IOCTL_NEXT_DEVICE or TCGETS, {B38400 opost isig icanon echo ...}) = 0
readlink("/proc/self/fd/0", "/dev/pts/1", 4095) = 10
getcwd("/home/lily/20150922/snoopy-2.4.4/util", 4097) = 38
getrlimit(RLIMIT_STACK, {rlim_cur=10240*1024, rlim_max=RLIM64_INFINITY}) = 0
getrlimit(RLIMIT_STACK, {rlim_cur=10240*1024, rlim_max=RLIM64_INFINITY}) = 0
write(1, "Message generated:\n", 19Message generated:
)    = 19
write(1, "\n", 1
)                       = 1
write(1, "[uid:1003 sid:8601 tty:/dev/pts/"..., 201[uid:1003 sid:8601 tty:/dev/pts/1 cwd:/home/lily/20150922/snoopy-2.4.4/util filename:/home/lily/20150922/snoopy-2.4.4/util/snoopy-test-output]: /home/lily/20150922/snoopy-2.4.4/util/snoopy-test-output
) = 201
write(1, "\n", 1
)                       = 1
socket(PF_LOCAL, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 3
connect(3, {sa_family=AF_LOCAL, sun_path="/dev/log"}, 10) = 0
sendto(3, "<86>snoopy[15009]: [uid:1003 sid"..., 219, MSG_DONTWAIT|MSG_NOSIGNAL, NULL, 0) = 219
close(3)                                = 0
write(1, "Message sent to output 'devlog()"..., 35Message sent to output 'devlog()'.
) = 35
write(1, "If Snoopy is already enabled on "..., 84If Snoopy is already enabled on your system, you should see two identical messages.
) = 84
write(1, "If you are testing Snoopy via LD"..., 105If you are testing Snoopy via LD_PRELOAD environmental variable, you will see another identical message.
) = 105
exit_group(0)                           = ?


_____________________________________________________________





and this is the user change to root and have snoopy output message


________________________________________________________________


[root@test util]#  strace /home/lily/20150922/snoopy-2.4.4/util/snoopy-test-output
execve("/home/lily/20150922/snoopy-2.4.4/util/snoopy-test-output", ["/home/lily/20150922/snoopy-2.4.4"...], [/* 22 vars */]) = 0
brk(0)                                  = 0x10c3000
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6861e68000
access("/etc/ld.so.preload", R_OK)      = 0
open("/etc/ld.so.preload", O_RDONLY)    = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=28, ...}) = 0
mmap(NULL, 28, PROT_READ|PROT_WRITE, MAP_PRIVATE, 3, 0) = 0x7f6861e67000
close(3)                                = 0
open("/usr/local/lib/libsnoopy.so", O_RDONLY) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0PQ\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=233989, ...}) = 0
mmap(NULL, 2149952, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f6861a3d000
mprotect(0x7f6861a4a000, 2093056, PROT_NONE) = 0
mmap(0x7f6861c49000, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xc000) = 0x7f6861c49000
close(3)                                = 0
munmap(0x7f6861e67000, 28)              = 0
open("/etc/ld.so.cache", O_RDONLY)      = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=36747, ...}) = 0
mmap(NULL, 36747, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f6861e5f000
close(3)                                = 0
open("/lib64/libpthread.so.0", O_RDONLY) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\340]\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=142640, ...}) = 0
mmap(NULL, 2212848, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f6861820000
mprotect(0x7f6861837000, 2097152, PROT_NONE) = 0
mmap(0x7f6861a37000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x17000) = 0x7f6861a37000
mmap(0x7f6861a39000, 13296, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f6861a39000
close(3)                                = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6861e5e000
open("/lib64/libdl.so.2", O_RDONLY)     = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\340\r\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=19536, ...}) = 0
mmap(NULL, 2109696, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f686161c000
mprotect(0x7f686161e000, 2097152, PROT_NONE) = 0
mmap(0x7f686181e000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x2000) = 0x7f686181e000
close(3)                                = 0
open("/lib64/libc.so.6", O_RDONLY)      = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0p\356\1\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=1921176, ...}) = 0
mmap(NULL, 3750152, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f6861288000
mprotect(0x7f6861412000, 2097152, PROT_NONE) = 0
mmap(0x7f6861612000, 20480, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x18a000) = 0x7f6861612000
mmap(0x7f6861617000, 18696, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f6861617000
close(3)                                = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6861e5d000
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6861e5c000
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6861e5b000
arch_prctl(ARCH_SET_FS, 0x7f6861e5c700) = 0
mprotect(0x7f6861612000, 16384, PROT_READ) = 0
mprotect(0x7f686181e000, 4096, PROT_READ) = 0
mprotect(0x7f6861a37000, 4096, PROT_READ) = 0
mprotect(0x7f6861e69000, 4096, PROT_READ) = 0
munmap(0x7f6861e5f000, 36747)           = 0
set_tid_address(0x7f6861e5c9d0)         = 15708
set_robust_list(0x7f6861e5c9e0, 24)     = 0
futex(0x7ffff588d6cc, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x7ffff588d6cc, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 1, NULL, 7f6861e5c700) = -1 EAGAIN (Resource temporarily unavailable)
rt_sigaction(SIGRTMIN, {0x7f6861825c60, [], SA_RESTORER|SA_SIGINFO, 0x7f686182f710}, NULL, 8) = 0
rt_sigaction(SIGRT_1, {0x7f6861825cf0, [], SA_RESTORER|SA_RESTART|SA_SIGINFO, 0x7f686182f710}, NULL, 8) = 0
rt_sigprocmask(SIG_UNBLOCK, [RTMIN RT_1], NULL, 8) = 0
getrlimit(RLIMIT_STACK, {rlim_cur=10240*1024, rlim_max=RLIM64_INFINITY}) = 0
brk(0)                                  = 0x10c3000
brk(0x10e4000)                          = 0x10e4000
open("/usr/local/etc/snoopy.ini", O_RDONLY) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=8175, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6861e67000
read(3, ";;; REQUIRED Section\n;\n[snoopy]\n"..., 4096) = 4096
read(3, "ilable=yes) Exclude log entries "..., 4096) = 4079
read(3, "", 4096)                       = 0
close(3)                                = 0
munmap(0x7f6861e67000, 4096)            = 0
fstat(1, {st_mode=S_IFCHR|0620, st_rdev=makedev(136, 1), ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6861e67000
write(1, "Configuration file is enabled: /"..., 57Configuration file is enabled: /usr/local/etc/snoopy.ini
) = 57
write(1, "Configuration file found.\n", 26Configuration file found.
) = 26
write(1, "Configuration file was parsed su"..., 43Configuration file was parsed sucessfully.
) = 43
getuid()                                = 0
getsid(0)                               = 8601
ioctl(0, SNDCTL_TMR_TIMEBASE or SNDRV_TIMER_IOCTL_NEXT_DEVICE or TCGETS, {B38400 opost isig icanon echo ...}) = 0
readlink("/proc/self/fd/0", "/dev/pts/1", 4095) = 10
getcwd("/home/lily/20150922/snoopy-2.4.4/util", 4097) = 38
getrlimit(RLIMIT_STACK, {rlim_cur=10240*1024, rlim_max=RLIM64_INFINITY}) = 0
getrlimit(RLIMIT_STACK, {rlim_cur=10240*1024, rlim_max=RLIM64_INFINITY}) = 0
write(1, "Message generated:\n", 19Message generated:
)    = 19
write(1, "\n", 1
)                       = 1
write(1, "[uid:0 sid:8601 tty:/dev/pts/1 c"..., 198[uid:0 sid:8601 tty:/dev/pts/1 cwd:/home/lily/20150922/snoopy-2.4.4/util filename:/home/lily/20150922/snoopy-2.4.4/util/snoopy-test-output]: /home/lily/20150922/snoopy-2.4.4/util/snoopy-test-output
) = 198
write(1, "\n", 1
)                       = 1
socket(PF_LOCAL, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 3
connect(3, {sa_family=AF_LOCAL, sun_path="/dev/log"}, 10) = 0
sendto(3, "<86>snoopy[15708]: [uid:0 sid:86"..., 216, MSG_DONTWAIT|MSG_NOSIGNAL, NULL, 0) = 216
close(3)                                = 0
write(1, "Message sent to output 'devlog()"..., 35Message sent to output 'devlog()'.
) = 35
write(1, "If Snoopy is already enabled on "..., 84If Snoopy is already enabled on your system, you should see two identical messages.
) = 84
write(1, "If you are testing Snoopy via LD"..., 105If you are testing Snoopy via LD_PRELOAD environmental variable, you will see another identical message.
) = 105
exit_group(0)                           = ?
+++ exited with 0 +++
[root@test util]#


_______________________________________________________________

Can u check what is it wrong?

**Comment by bostjan** [comment]
@clover1983 : HI.

This part of the code is present in both traces:
socket(PF_LOCAL, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 3
connect(3, {sa_family=AF_LOCAL, sun_path="/dev/log"}, 10) = 0
sendto(3, "<86>snoopy[15009]: [uid:1003 sid"..., 219, MSG_DONTWAIT|MSG_NOSIGNAL, NULL, 0) = 219
close(3) = 0

This does the actual transmission of message to /dev/log socket, which is then picked up by your syslog daemon.

There could be some things messing with your snoopy output:
- funky syslog configuration (dropping messages with too low priority, storing in other file)
- apparmor/selinux/systemd intercepting snoopy messages and discarding them
Do check these and report back if possible. Maybe reconfigure your syslog to route all messages (regardles of facility and priority) into a dedicated file and check for snoopy messages there.

Are you also using RHEL 6.6?

b.

**Comment by bostjan** [comment]
No reply for 3 months, closing.
